### PR TITLE
[ISSUE-4184] Separate offline and state plugins

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### 🎯 Goal

Separate Offline and State plugin

### 🛠 Implementation details

- Deleted the Composite Listeners
- Adding a `withPlugins` methods and using it to add 2 plugins, state and offline. 

### 🧪 Testing

Test the SDK as a whole.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- ~[ ] Comparison screenshots added for visual changes~
- ~[ ] Affected documentation updated (KDocs, docusaurus, tutorial)~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF
giphy (3)
=D

### CODE REVIEW

File: stream-chat-android-offline/build.gradle [hunk 1]
```diff
@@ -57,7 +57,6 @@ tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
 
 dependencies {
     api project(":stream-chat-android-client")
-    api project(":stream-chat-android-state")
```
Comment (JcMinarro) [inline stream-chat-android-offline/build.gradle:60]: :partying_face:

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt [hunk 2]
```diff
@@ -2830,6 +2830,10 @@ internal constructor(
             pluginFactories.add(pluginFactory)
         }
 
+        public fun withPlugins(vararg pluginFactoryArgs: PluginFactory): Builder = apply {
```
Comment (JcMinarro) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:2833]: Do we need the `withPlugin(pluginFactory: PluginFactory)`  method?
As far as this new one accepts a `vararg` arguments, we can provide only this one.

File: stream-chat-android-offline/build.gradle (hunk 1 above)
Comment (leandroBorgesFerreira) [reply stream-chat-android-offline/build.gradle:60]: \o/

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt [hunk 3]
```diff
@@ -2820,14 +2820,8 @@ internal constructor(
...
-        /**
-         * Adds a plugin factory to be used by the client.
-         * @see [PluginFactory]
-         *
-         * @param pluginFactory The factory to be added.
-         */
-        public fun withPlugin(pluginFactory: PluginFactory): Builder = apply {
-            pluginFactories.add(pluginFactory)
+        public fun withPlugins(vararg pluginFactoryArgs: PluginFactory): Builder = apply {
+            pluginFactories.addAll(pluginFactoryArgs)
```
Comment (JcMinarro) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:2824]:
```suggestion
        * Adds plugins factory to be used by the client.
         * @see [PluginFactory]
         *
         * @param pluginFactories The factories to be added.
         */
        public fun withPlugins(vararg pluginFactories: PluginFactory): Builder = apply {
            this.pluginFactories.addAll(pluginFactories)
```

**Review by JcMinarro — APPROVED** [review]

**Review by filbabic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs


Works on the UI side, however it's really hard to test the SDK "broadly" as we should probably run a regression test suite.

**Pull request merged**